### PR TITLE
Performance improvements

### PR DESCRIPTION
**Pull request by Thurman1776-**
While profiling the app I've noticed there are some noticeable overhead when letting [Yams](https://github.com/jpsim/Yams) do its work. Right now, the same approach is being takes as in [here](https://github.com/Thurman1776-/Uncomplect-us/blob/master/Mac-App/Backend/Backend/FileSystem/FindProjectOutputDirectories.swift#L102-L139) but with slight modifications to potentially make it reusable. Maybe later replace the old one with this generic version? 

Duplicates are now being skipped that drastically reduces the number of items to display.

### CODE REVIEW

**Comment by Thurman1776-** [comment]
Hey @emrepun  

Would you like to revive this [PR](https://github.com/Thurman1776-/Uncomplect-us/pull/3) & give a second shot based on the discussion we had online regarding [this API](https://developer.apple.com/documentation/foundation/nsprocessinfo/1415622-processorcount?language=objc)? 

I was experimenting with the idea for a bit & while it seemed promising, I got some random crashes (different from what you experienced) but I believe that crash might be related of how the current implementation is done.

File: Mac-App/Backend/Backend/Shared/Helpers /ConcurrentWork.swift [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+) -> [O] {
+    let middleIndex: Double = floor(Double(collection.count / 2))
+    let firstChunk = collection.prefix(upTo: Int(middleIndex))
+    let secondChunk = collection.suffix(from: Int(middleIndex))
+
+    var allItems: Set<O> = []
+    var firstChunkItems: Set<O> = []
+    var secondChunkItems: Set<O> = []
+
+    let operation = OperationQueue()
```
Comment (emrepun) [inline Mac-App/Backend/Backend/Shared/Helpers /ConcurrentWork.swift]: I think renaming this to `operationQueue` or `queue` would be better.

**Review by emrepun — CHANGES_REQUESTED** [review]

**Comment by emrepun** [comment]
> Maybe later replace the old one with this generic version?

I think it would be really nice @Thurman1776-  👍

File: Mac-App/Backend/Backend/Shared/Helpers /ConcurrentWork.swift (hunk 1 above)
Comment (Thurman1776-) [reply Mac-App/Backend/Backend/Shared/Helpers /ConcurrentWork.swift]: True

**Comment by emrepun** [comment]
> Hey @emrepun
> 
> Would you like to revive this [PR](https://github.com/Thurman1776-/Uncomplect-us/pull/3) & give a second shot based on the discussion we had online regarding [this API](https://developer.apple.com/documentation/foundation/nsprocessinfo/1415622-processorcount?language=objc)?
> 
> I was experimenting with the idea for a bit & while it seemed promising, I got some random crashes (different from what you experienced) but I believe that crash might be related of how the current implementation is done.

Yes I'd like to actually. Will open the PR and try to experiment it further asap! @Thurman1776-

File: Mac-App/Backend/Backend/Shared/Helpers /ConcurrentWork.swift (hunk 1 above)
Comment (Thurman1776-) [reply Mac-App/Backend/Backend/Shared/Helpers /ConcurrentWork.swift]: https://github.com/Thurman1776-/Uncomplect-us/pull/16/commits/365e4e4c500db1c463c117493edb21cb9688c8f9

**Pull request merged**